sympy/printing/pretty/pretty.py

```diff
@@ -1811,7 +1811,7 @@ def _print_DiracDelta(self, e):
             return self._print_Function(e)
 
     def _print_expint(self, e):
-        if e.args[0].is_Integer and self._use_unicode:
+        if _opportunistic_subscripter(self._print(e.args[0])) and self._use_unicode:
             return self._print_Function(Function('E_%s' % e.args[0])(e.args[1]))
```

Comment:
Following up a bit on @sylee957's comment...  We need to be careful here: you can't ask permission about the `_print(args[0])` but then use the the non-printed `args[0]`.

Probably we can think of several fixes...

Alternatively: "zen of python" says "better to ask forgiveness than permission".  So another possibility is to code it in a way that this possibility is avoided?

Reply:
I also did write a fix for this ! (just forgot to comment it here)
Defined a variable called `subscript` that is used by both `is_subscriptable_in_unicode()` and `self._print_Function`